src/main/java/io/dexi/config/Config.java

```diff
@@ -0,0 +1,187 @@
+package io.dexi.config;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.FileBasedConfiguration;
+import org.apache.commons.configuration2.YAMLConfiguration;
+import org.apache.commons.configuration2.builder.FileBasedConfigurationBuilder;
+import org.apache.commons.configuration2.builder.ReloadingFileBasedConfigurationBuilder;
+import org.apache.commons.configuration2.builder.fluent.Parameters;
+import org.apache.commons.configuration2.builder.fluent.PropertiesBuilderParameters;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.lang.StringUtils;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ *  This class supports reading a "hierarchical" configuration file with sections like:
+ *
+ *  <b>YAML example</b>
+ *  <pre>
+ *  {@code
+ *     dexi:
+ *       baseUrl: http://localhost:3000/api/
+ *       apiKey: super-secret-key
```

Comment:
missing accountId property

Reply:
Actually it looks like it's called "account" - not sure what is better

Reply:
This is just an *example* of what configuration looks like. I actually removed `account` on purpose because I didn't think it belonged it this type of configuration. Whether that is true or not doesn't really change that this is just an example of configuration.

Reply:
It is needed for the Dexi API SDK to work - so if it's not there it'll not work

Reply:
I think we are talking about two different things. I have implemented general functionality for configuration - not specific to the Dexi API SDK. The general functionality can then be used to get specific configuration that works to be able to access the Dexi API SDK. I did not know I was supposed to do that in this task.